src/wasm/wasm-binary.cpp

```diff
@@ -219,16 +219,33 @@ void WasmBinaryWriter::writeTypes() {
   if (indexedTypes.types.size() == 0) {
     return;
   }
+  // Count the number of recursion groups
+  std::optional<RecGroup> lastGroup;
+  size_t numGroups = 0;
+  for (auto type : indexedTypes.types) {
+    auto currGroup = type.getRecGroup();
+    numGroups += lastGroup != currGroup;
+    lastGroup = currGroup;
+  }
   BYN_TRACE("== writeTypes\n");
   auto start = startSection(BinaryConsts::Section::Type);
-  o << U32LEB(indexedTypes.types.size());
+  o << U32LEB(numGroups);
```

Comment:
In the non-hybrid case, this is just the number of types, is that right? (one type per "rec group") If so, it might be good to clarify that.